array/DNAm/preprocessing/checkColnamesFunction.r

```diff
@@ -0,0 +1,51 @@
+
+
+# Checks sample sheet columns are formatted correctly
+checkColnames <- function(sampleSheet, exp_cols, type, verbose=T){
+	metrics <- list()
+	
+	all_pres <- all(exp_cols %in% colnames(sampleSheet))
+	metrics[['allPresent']] <- all_pres
+	
+	if(all_pres){
+		if(verbose){print(paste(type, "column names: All present"))}
+	}else{
+	
+		if(verbose){print(paste(type, "column names: Not all present"))}
+		
+		#a) find correct cols
+		pres <- exp_cols[exp_cols %in% colnames(sampleSheet)]
+		if(length(pres)>0){
+			if(verbose){print(c("Identical colnames:",pres))}
+		}
+		metrics[['presentIDs']] <- pres
+		
+		#b) find absent cols
+		ab <- exp_cols[exp_cols %ni% colnames(sampleSheet)]
+		if(verbose){print(c("Absent colnames:",ab))}
+		metrics[['absentIDs']] <- ab
+		
+		#c) find close matches
+		near <- exp_cols[amatch(colnames(sampleSheet), exp_cols, maxDist = 2)] # maxDist = number of characters difference

```

Comment:
Re your first point, I see what you mean. But since it is important the column names are written with the correct separator (underscore), a column called Cell.Type might incorrectly match to Cell_Type because we've stripped the separator. What do you think?

Re your second point, hopefully this answers it but let me know if not. 
Yes the sample sheet could have 2 column names that are similarly named for example Cell_type and Cell_Type. 
If "Cell_Type" is the required column name then this will match perfectly to one of the columns and will be flagged in `pres`. When it comes to the near matches, amatch picks up both near matches and exact matches so both Cell_Type and Cell_type will be flagged again here. 
Since in this section we are interested in near matches I figured that we would want to remove from the printed output any cases of exact matches (see variable `equal`). 
And if this happens to be a case where there is an exact match AND a near match (such as Cell_type and Cell_Type), I don't think we need to flag anything wrong here because it has met the criteria of having the required column name (Cell_Type). Therefore I find and remove them both (see variable `other`). This is what I attempt to do in lines 34-39.

Reply:
Oh ok, I think my understanding of the original problem was wrong.

I though you were trying to account for column names being inconsistent between sample sheets. So for example, in one sample sheet you might have `Cell_Type` whilst in another you might have `cell.type` (where these are actually the same column, displaying the same kind of information, just with different names). 

That's why I made those suggestions. As `Cell_Type` and `cell.type` have a distance of 3 from one another. If you remove separators and make them lower case they are the same (so you can be more strict on the maxDist). 

My bad.

Reply:
Ah I see!